File: lab/javascript/matematica_e_operadores/permutacoes.js
function fatorial(n){
    let m = 1;
    for (let i=n; i > 0; i--){
        m *= i;
    }
    return m;
}

function permutacao(n){
    return fatorial(n);
}

function permutacao_parcial(n, k){
    return Math.floor(fatorial(n) / fatorial(n - k));
}

console.log(permutacao(4));
console.log(permutacao_parcial(5, 3));

File: lab/python/matematica_e_operadores/permutacoes.py
def fatorial(n):
    m = 1
    for i in range(n, 0, -1):
        m *= i
    return m


def permutacao(n):
    return fatorial(n)


def permutacao_parcial(n, k):
    return fatorial(n) // fatorial(n - k)


print(permutacao(4))
print(permutacao_parcial(5, 3))
